storage_controller/migrations/2024-07-23-191537_create_metadata_health/up.sql

```diff
@@ -0,0 +1,8 @@
+CREATE TABLE metadata_health (
+  tenant_id VARCHAR NOT NULL,
+  shard_number INTEGER NOT NULL,
+  shard_count INTEGER NOT NULL,
+  PRIMARY KEY(tenant_id, shard_number, shard_count),
```

Comment:
Should this be a foreign key into the `tenant_shards` table? It would allow for failing inserts for tenant shards that are not in the parent table and cascading deletes.

Reply:
Great call! Added in a93976255814d3f96cd719b9c8a9032d2226131a